nixos/modules/services/web-servers/apache-httpd/default.nix

```
@@ -36,10 +36,13 @@ let
   dependentCertNames = unique (map (hostOpts: hostOpts.certName) acmeEnabledVhosts);
 
   mkListenInfo = hostOpts:
+    let
+      addrs = if hostOpts.listenAddresses != [] then hostOpts.listenAddresses else [ "*" ];
```

```suggestion
      addrs = hostOpts.listenAddresses;
```

In conjunction with option change below.
(and you can inline `addrs` of course)